be/src/vec/exec/format/table/iceberg_reader.cpp

```diff
@@ -136,6 +138,32 @@ Status IcebergTableReader::init_reader(
 
     return status;
 }
+Status IcebergTableReader::init_reader_for_orc(
```

Comment:
Can we abstract the abstract class for the iceberg reader, using parquet/orc as the concrete implementation? If there are reuse functions between two formats, it can be moved to the parent class. The code format like `if-else` often increases the difficulty of later maintenance and modification.